src/bindings/python/flux/asyncio/selector.py

```diff
@@ -0,0 +1,153 @@
+    """
+    _ = watcher
+    reactor = handle.get_reactor()
+    handle.reactor_stop(reactor)
+
+    # TimedWatcher won't know a file descriptor, just needs to stop the reactor
+    if not args:
+        return
+    selector = args["select"]
+    selector_event = get_selector_event(events)
+    selector.ready.add((args["key"], selector_event & args["key"].events))
+
+
+def get_selector_event(events):
+    """
+    Given an int, return the corresponding flux identifier.
+    """
+    event = 0
+    if events & FLUX_POLLIN:
+        event |= selectors.EVENT_READ
+    if events & FLUX_POLLOUT:
+        event |= selectors.EVENT_WRITE
+    return event
+
+
+def get_flux_event(events):
+    """
+    Given an int, return the corresponding selector identifier
+    """
+    event = 0
+    if events & selectors.EVENT_READ:
+        event |= FLUX_POLLIN
+    if events & selectors.EVENT_WRITE:
+        event |= FLUX_POLLOUT
+    return event
+
+
+class FluxSelector(selectors._BaseSelectorImpl):  # pylint: disable=protected-access
+    """
+    A Flux selector supports registering file objects to be monitored for
+    specific I/O events (for Flux).
+    """
+
+    def __init__(self, handle):
+        super().__init__()
+        self.handle = handle
+        self.ready = set()
+        self._watchers = {}
+
+    def register(self, fileobj, events, data=None):
+        """
+        Register a new file descriptor event.
+        """
+        key = super().register(fileobj, events, data)
+        watcher = FDWatcher(
+            self.handle,
+            fileobj,
+            get_flux_event(events),
+            stop_callback,
+            args={"key": key, "select": self},
+        )
+        watcher.start()
+        self._watchers[fileobj] = watcher
+        return key
+
+    def unregister(self, fileobj):
+        """
+        Remove the key and the watcher.
+        """
+        try:
+            key = self._fd_to_key.pop(self._fileobj_lookup(fileobj))
+            self._watchers[key.fileobj].stop()
+        except KeyError:
+            raise KeyError("{!r} is not registered".format(fileobj)) from None
+        return key
+
+    def select(self, timeout=None):
+        """
+        Perform the actual selection, until some monitored file objects are
+        ready or a timeout expires.
+        Parameters:
+        timeout -- if timeout > 0, this specifies the maximum wait time, in
+                   seconds, waited for by a flux Timed Watcher
+                   if timeout <= 0, the select() call won't block, and will
+                   report the currently ready file objects
+                   if timeout is None, select() will block until a monitored
+                   file object becomes ready
+        Returns:
+        list of (key, events) for ready file objects
+        `events` is a bitwise mask of EVENT_READ|EVENT_WRITE
+        """
+        # Store the watcher as data
+        reactor = self.handle.get_reactor()
```

Comment:
This one needs the flags for the second argument, and this is also how I see it used elsewhere, so I think it would be OK to leave like this to be consistent.

Reply:
You could pass the flags as a keyword argument `(flags=...)` but OK.

Reply:
okay, happy to do that too!

Reply:
Actually I take it back - I had tried this and it doesn't work. If you provide flags, it gets angry that the argument isn't correct. If I look at all other use cases in the code, two args are provided. If I stray from that:

```python
WrongNumArguments: 
The wrong number of arguments has been passed to wrapped C function:
Expected 2 arguments, received 0
Name: flux_reactor_run
C signature: int(*)(struct flux_reactor *, int)
Arguments: []
Handle type: <ctype 'struct flux_handle_struct *'>
```
So I probably put the reactor and flags there as I did because that's how I got it to work!

Reply:
Going to resolve this - unresolve if there is another way you want to try!